### ISSUE TITLE
[DX] Inconsistent application of case sensitivity

### ISSUE THREAD

**Issue by databyjp** [weaviate/weaviate#5789]
TL;DR: Weaviate treats case sensitivity differently in collection names depending on operation, leading to confusion. 

**Scenario:**
If a user creates a collection called `My_wiki`, and subsequently tries to create `My_Wiki`, Weaviate will throw due to 'class already exists: found similar class "My_wiki"'
But, attempting to delete `My_Wiki` will not resolve it, because the same similarity (`My_Wiki` vs `My_wiki`) does not cause deletion of the similar collection.

This can be confusing to the user, as they are blocked from creating `My_Wiki`, but deleting a collection with the same name and creating a new one is not allowed. 

What should the desired behaviour be?

**Comment by kavirajk** [comment]
I faced similar confusion today :) 

e.g:
`GET http://localhost:8080/v1/schema/Question` - OK
`GET http://localhost:8080/v1/schema/question` - 404

curious what stops us to make `collection-name` case insensitive everywhere?
